nucypher/blockchain/eth/token.py

```diff
@@ -590,28 +607,102 @@ def _crash_gracefully(self, failure=None) -> None:
     def handle_working_errors(self, *args, **kwargs) -> None:
         failure = args[0]
         if self._abort_on_error:
-            self.log.critical('Unhandled error during node work tracking. {failure!r}',
+            self.log.critical(f'Unhandled error during node work tracking. {failure!r}',
                               failure=failure)
             reactor.callFromThread(self._crash_gracefully, failure=failure)
         else:
-            self.log.warn('Unhandled error during work tracking: {failure.getTraceback()!r}',
+            self.log.warn(f'Unhandled error during work tracking: {failure.getTraceback()!r}',
                           failure=failure)
 
-    def __check_work_requirement(self) -> bool:
+    def __work_requirement_is_satisfied(self) -> bool:
         # TODO: Check for stake expiration and exit
         if self.__requirement is None:
             return True
-        try:
-            r = self.__requirement()
-            if not isinstance(r, bool):
-                raise ValueError(f"'requirement' must return a boolean.")
-        except TypeError:
-            raise ValueError(f"'requirement' must be a callable.")
+        r = self.__requirement()
+        if not isinstance(r, bool):
+            raise ValueError(f"'requirement' must return a boolean.")
         return r
 
+    @property
+    def pending(self) -> Dict[int, HexBytes]:
+        return self.__pending.copy()
+
+    def __track_pending_commitments(self) -> bool:
+
+        worker_address = self.worker.transacting_power.account  # FIXME: Why is worker_address private in Worker?
+        tx_count_pending = self.client.get_transaction_count(account=worker_address, pending=True)
+        tx_count_latest = self.client.get_transaction_count(account=worker_address, pending=False)
+
+        txs_in_mempool = tx_count_pending - tx_count_latest
+
+        if txs_in_mempool > len(self.__pending):  # We're not tracking all pending TXs
+            tx_tracker_is_ok = False
+        elif txs_in_mempool < len(self.__pending):  # Our tracking is somehow outdated
+            tx_tracker_is_ok = False  # TODO: Not sure what to do in this case, but let's do this for the moment
+        else:
+            tx_tracker_is_ok = True
+
+        if not tx_tracker_is_ok:
+            # If we detect there's a mismatch between the number of tracked and existing pending transactions,
+            # we create a special pending TX that accounts for this.
+            self.__pending[0] = UNTRACKED_PENDING_TRANSACTION
+            return True
+
+        if not self.__pending:
+            return False
+
+        unmined_transactions = list()
```

Comment:
Switching the order of the above and below code block allows the test to pass, since `self.__pending` needs to be pruned *first* in order to remain consistent

Reply:
:heavy_check_mark: